crates/corro-agent/src/agent/handlers.rs

```diff
@@ -863,15 +863,22 @@ pub async fn handle_changes(
             continue;
         }
 
-        // drop items when the queue is full.
+        // drop old items when the queue is full.
         if queue.len() > max_queue_len {
+            let change = queue.pop_front();
```

Comment:
Should this be `pop_back`? We're popping from the front in the rest of the logic.

Reply:
Ah yes. We are processing received changes in order since biggest versions sent first.